clients/go/test/integration_test.go

```diff
@@ -378,6 +365,123 @@ func (s *integrationTestSuite) TestFailJob() {
 	}
 }
 
+func (s *integrationTestSuite) TestStreamingJobWorker() {
+	// given
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	taskType := uuid.NewString()
+	bpmn, err := s.readBpmnWithCustomJobType("testdata/service_task.bpmn", taskType)
+	s.NoError(err)
+
+	deployment, err := s.client.NewDeployResourceCommand().AddResource(bpmn, "service_task.bpmn").Send(ctx)
+	if err != nil {
+		s.T().Fatal(err)
+	}
+
+	deployedResource := deployment.GetDeployments()[0]
+	s.NotNil(deployedResource)
+
+	process := deployedResource.GetProcess()
+	s.NotNil(process)
+
+	// when
+	// the stream will be closed by the deferred context cancellation
+	ctx, cancel = context.WithCancel(context.Background())
+	defer cancel()
+
+	jobsChan := make(chan entities.Job)
+	defer close(jobsChan)
+
+	workerName := uuid.New().String()
+	jobWorker := s.client.NewJobWorker().
+		JobType(taskType).
+		Handler(func(client worker.JobClient, job entities.Job) {
+			jobsChan <- job
+		}).
+		Timeout(time.Minute * 5).
+		PollInterval(1 * time.Hour). // poll very slowly to make sure we get our jobs from streaming
+		Name(workerName).
+		StreamEnabled(true).
+		RequestTimeout(time.Duration(1) * time.Second).
+		Open()
+	defer jobWorker.Close()
+
+	// Await until the stream is created on the broker
+	streamExists := s.awaitJobStreamExists(workerName)
+	s.True(streamExists, "Expected remote stream to exist on broker after 5 seconds, but none yet exist")
+
+	// create two PIs and expect two jobs
+	_, err = s.client.NewCreateInstanceCommand().ProcessDefinitionKey(process.GetProcessDefinitionKey()).Send(ctx)
+	s.NoError(err)
+	_, err = s.client.NewCreateInstanceCommand().ProcessDefinitionKey(process.GetProcessDefinitionKey()).Send(ctx)
+	s.NoError(err)
+
+	// then - expect two jobs
+	jobs := make([]entities.Job, 0)
+	for i := 0; i < 2; i++ {
+		job, ok := <-jobsChan
+		if ok {
+			jobs = append(jobs, job)
+		} else {
+			break
+		}
+	}
+	s.Len(jobs, 2, "Expected to receive 2 jobs")
```

Comment:
❓ couldn't we do this directly in the handler?

Reply:
Maybe? I wasn't sure about the thread safety of modifying a slice from different go routine :s